full/src/main/java/apoc/couchbase/document/CouchbaseJsonDocument.java

```diff
@@ -42,17 +45,33 @@
   public Map<String, Object> mutationToken;
 
   /**
-   * The content of the {@link Document}.
+   * The content of the {@link GetResult}.
    */
   public Map<String, Object> content;
 
-  public CouchbaseJsonDocument(JsonDocument jsonDocument) {
-    this.id = jsonDocument.id();
-    this.expiry = jsonDocument.expiry();
-    this.cas = jsonDocument.cas();
-    this.mutationToken = CouchbaseUtils.convertMutationTokenToMap(jsonDocument.mutationToken());
-    this.content = (jsonDocument.content() != null) ? this.content = jsonDocument.content().toMap()
-        : new HashMap<String, Object>();
+  public CouchbaseJsonDocument(GetResult getResult, String id) {
+    this.id = id;
+    this.expiry = getResult.expiryTime().orElse(Instant.ofEpochMilli(0)).toEpochMilli();
+    this.cas = getResult.cas();
+    this.mutationToken = null;
+    this.content = getResult.contentAsObject().toMap();
+  }
+
+  public CouchbaseJsonDocument(MutationResult mutationResult, String id, Collection collection) {
+
+    GetResult getResult = collection.exists(id).exists() ? collection.get(id) : null;
+
+    this.id = id;
+    this.cas = mutationResult.cas();
```

Comment:
perché non si può prendere `getResult.cas()`?

io inoltre invocherei il costruttore sopra da qui

Reply:
fai tutte le variabili final

Reply:
`getResult.cas()` si può usare per l'update o l'insert, mentre sul `delete` non essendoci `getResult` non si può usare e bisogna mettere `mutationResult.cas();`. Chiamare il costruttore di sopra essendoci `this.mutationToken = null;` non si può fare, a meno di mettere un'altro costruttore intermedio o cose così, ma mi sembra poco ottimale come cosa

Reply:
@vga scusa fai un terzo costruttore che prende mutationToken come argomento no? e poi chiama questo (appena creato) dagli altri